src/spikeinterface/core/testing.py

```diff
@@ -73,22 +68,15 @@ def check_sortings_equal(SX1, SX2):
             train2 = np.sort(SX2.get_unit_spike_train(id, segment_index=segment_idx))
             assert np.array_equal(train1, train2)
 
-# def check_sorting_properties_features(SX1, SX2):
-#     # check properties
-#     print(SX1.__class__)
-#     print('Properties', sorted(SX1.get_shared_unit_property_names()), sorted(SX2.get_shared_unit_property_names()))
-#     assert sorted(SX1.get_shared_unit_property_names()) == sorted(SX2.get_shared_unit_property_names())
-#     for prop in SX1.get_shared_unit_property_names():
-#         for u in SX1.get_unit_ids():
-#             if not isinstance(SX1.get_unit_property(u, prop), str):
-#                 assert np.allclose(np.array(SX1.get_unit_property(u, prop)),
-#                                    np.array(SX2.get_unit_property(u, prop)))
-#             else:
-#                 assert SX1.get_unit_property(u, prop) == SX2.get_unit_property(u, prop)
-#     # check features
-#     print('Features', sorted(SX1.get_shared_unit_spike_feature_names()), sorted(SX2.get_shared_unit_spike_feature_names()))
-#     assert sorted(SX1.get_shared_unit_spike_feature_names()) == sorted(SX2.get_shared_unit_spike_feature_names())
-#     for feat in SX1.get_shared_unit_spike_feature_names():
-#         for u in SX1.get_unit_ids():
-#             assert np.allclose(np.array(SX1.get_unit_spike_features(u, feat)),
-#                                np.array(SX2.get_unit_spike_features(u, feat)))
+    if check_properties:
+        check_extractor_properties_equal(RX1, RX2)
+
+
+def check_extractor_properties_equal(EX1, EX2) -> None:
+    assert np.array_equal(sorted(EX1.get_annotation_keys()), sorted(EX2.get_annotation_keys()))
+    assert np.array_equal(sorted(EX1.get_property_keys()), sorted(EX2.get_property_keys()))
+
+    for annotation_name in EX1.get_annotation_keys():
+        assert EX1.get_annotation(annotation_name) == EX2.get_annotation(annotation_name)
+    for property_name in EX1.get_property_keys():
+        np.assert_array_equal(EX1.get_property(property_name), EX2.get_property(property_name))
```

Comment:
I would split properties and annotations honestly. For example, a `bandpass_filter` will add the `is_filtered=True` annotation, but properties will be the same!

Reply:
True, done :)